client/src/worldgen.rs

```diff
@@ -256,6 +257,64 @@ impl ChunkParams {
         }
     }
 
+    fn generate_road_support(
+        &self,
+        center: na::Vector3<f64>,
+        coords: na::Vector3<u8>,
+    ) -> Option<Material> {
+        let plane = -Plane::from(Side::B);
+        let horizontal_distance = plane.elevation(center, self.chunk);
+
+        if horizontal_distance < 0.3 {
+            let mat = self
+                .generate_trussing(coords)
+                .unwrap_or_else(|| Material::WoodPlanks);
+            if mat == Material::Void {
+                None
+            } else {
+                Some(mat)
+            }
+        } else {
+            None
+        }
+    }
+
+    ///carve out a truss-shaped hole
+    fn generate_trussing(&self, coords: na::Vector3<u8>) -> Option<Material> {
```

Comment:
```suggestion
    fn trussing_at(&self, coords: na::Vector3<u8>) -> bool {
```
If we're just using this to determine the shape, rather than any material variation, let's use a more specific return type.